lib/semian/protected_resource.rb

```diff
@@ -34,6 +34,7 @@ def acquire_circuit_breaker(scope, adapter, resource)
         yield self
       else
         @circuit_breaker.acquire(resource) do
+          Semian.notify(:success, self, scope, adapter) if @bulkhead.nil?
```

Comment:
Would it make sense to delete the two current `Semian.notify` calls and put just one in the main `acquire` method?

Reply:
`acquire_bulkhead` adds the `wait_time`, `Semian.notify(:success, self, scope, adapter, wait_time)`. Because of that I don't think it's possible to add only one `Semian.notify` :(

Reply:
Why not use `wait_time || 0` in the common codepath?

Reply:
`wait_time` is added here: https://github.com/Shopify/semian/blob/d74c809554c33e452310e91bcbbf05ae07dd7aa6/lib/semian/protected_resource.rb#L50-L53

I don't think I could access `wait_time` from the common code path without adding a bit of complexity. Maybe I'm missing something, let me know.

Reply:
Right, you could `yield self, wait_time`, this seems not overly complex to me.

Reply:
I'm lost 🤦‍♂️ I don't know why I did not think about that. Done!